### PR TITLE
fix `AUTH` failed: ERR Client sent AUTH, but no password is set

### PR DESCRIPTION
**Pull request by hncg**
for example:
in laravel or other projects  .env file

```
REDIS_HOST=redis
REDIS_PASSWORD=
REDIS_PORT=6379
```

### CODE REVIEW

File: src/Connection/Factory.php [hunk 1]
```diff
@@ -173,7 +173,7 @@ protected function prepareConnection(NodeConnectionInterface $connection)
     {
         $parameters = $connection->getParameters();
 
-        if (isset($parameters->password)) {
+        if (empty($parameters->password)) {
```
Comment (mikebranderhorst) [inline src/Connection/Factory.php]: if ( ! empty($parameters->password)) {

**Comment by poisa** [comment]
@hncg Thank you! This has been driving me insane with Laravel until I saw your PR. I hope this gets merged ASAP!

**Comment by the-veloper** [comment]
`REDIS_PASSWORD` should be set to 'null' instead of nothing.
Ex:
```
REDIS_HOST=redis
REDIS_PASSWORD=null
REDIS_PORT=6379
```

**Comment by hncg** [comment]
Yes, but we shouldn’t  expect that everyone knows exactly how to set.
@the-veloper

**Comment by nrk** [comment]
Fixed for v1.1 in 31baa9b (will be included in v1.1.4).

**Pull request closed without merging**